core-api/src/main/java/com/absmartly/sdk/Context.java

```diff
@@ -149,7 +150,12 @@ public Context waitUntilReady() {
 		if (data_ == null) {
 			final CompletableFuture<Void> future = readyFuture_; // cache here to avoid locking
 			if (future != null && !future.isDone()) {
-				future.join();
+				future.whenComplete(new BiConsumer<Void, Throwable>() {
```

Comment:
I don't think this should be here for three reasons:
1. It seems like an implementation detail of the caching mechanism.
2. What if the user never calls waitUntilReady? i.e. the user may create the context with an already retrieved `contextData`.
3. It could be done independently when the `eventHandler` is created, or instead, we could have a notification-like method in the interface, for example `contextBecameReady()` or something like that, that is called on the `ContextEventHandler` interface when it gets ready (not just here).

Reply:
Did a change here, check if this place I put the code can have some side affect I not see.